omod/src/main/webapp/resources/scripts/flowsheet.js

```diff
@@ -53,8 +55,16 @@
         requireEncounter = reqEnc;
     };
 
-    flowsheet.addFlowsheet = function(index, formName, encounterIds) {
-        flowsheets.push(new Flowsheet(index, formName, encounterIds));
+    flowsheet.setEncounterTypeIds = function(encounterTypes) {
+      encounterTypeIds = encounterTypes;
```

Comment:
Same thing here, I think this is a encounterIdToEncounterTypeUuid map?  Would be good to add a comment here like "used because we only allow editing encounters that have the same encounter type as the form".

Reply:
Thanks @mogoodrich . I have addressed all your comments above. Once the build completes I will merge this PR. Thanks.

Reply:
Cool, thanks @cioan , yeah, go for it...